src/core/checkedint.d

```diff
@@ -517,3 +519,20 @@ unittest
     assert(mulu(0UL, 0UL, overflow) == 0);
     assert(overflow);                   // sticky
 }
+
+
+private:
+
+pragma(inline, true)
+@property size_t _signBit(T)(in T n)
+if(is(T == int) || is(T == long))
+{
+    return cast(size_t) (n >>> T.sizeof * 8 - 1);
```

Comment:
Many compilers generate very good code for `bool b = a < b;`. I suggest that instead of the shift. And besides, this function is a bit peculiar in returning all 0's or all 1's.